proxies/live/apiproxy/targets/ers-target.xml

```diff
@@ -89,19 +89,38 @@
                     <Condition>(request.header.echo)</Condition>
                 </Step> {% endif %} <Step>
                     <!--This should always be the last Step - as it is just before the message is sent - so the initial request stays intact for as long as possible.
-                The Swapping of the Request Headers converts X-Correlation-ID to NHSD-Correlation-ID before sending to backend. -->
+            The Swapping of the Request Headers converts X-Correlation-ID to NHSD-Correlation-ID before sending to backend. -->
                     <Name>AssignMessage.Swap.CorrelationHeader</Name>
                 </Step></Request>
             <Response/>
         </Flow>
-        <!-- App restricted not yet supported -->
         <Flow name="app-restricted-flow">
             <Condition>(accesstoken.auth_type == "app")</Condition>
-            <Request>
-                <Step>
-                    <Name>RaiseFault.403Forbidden</Name>
-                </Step>
-            </Request>
+            <Request><!-- reject if headers that are overriden as part of app rectricted call are provided--><Step>
+                    <Name>RaiseFault.400BadRequest</Name>
+                    <Condition>(request.header.x-ers-ods-code != null)</Condition>
+                </Step> <Step>
+                    <Name>RaiseFault.400BadRequest</Name>
+                    <Condition>(request.header.x-ers-business-function != null)</Condition>
+                </Step> <Step>
+                    <Name>RaiseFault.400BadRequest</Name>
+                    <Condition>(request.header.x-ers-user-id != null)</Condition>
+                </Step> <Step>
+                    <Name>AssignMessage.Set.nhsd-ers-ods-code-header-app-restricted</Name>
+                </Step> <Step>
+                    <Name>AssignMessage.Set.nhsd-ers-business-function-header-app-restricted</Name>
+                </Step> <Step>
+                    <Name>AssignMessage.Set.x-ers-user-id-header-app-restricted</Name>
+                </Step><Step>
+                    <Name>AssignMessage.Remove.x-request-id-header</Name>
+                </Step> {% if ALLOW_ECHO_TARGET | default(false) == true %} <Step>
+                    <Name>AssignMessage.SetEchoTarget</Name>
+                    <Condition>(request.header.echo)</Condition>
```

Comment:
As per this line here which checks for the presence of the header not caring about the value